### ISSUE TITLE
CountryPresenter must not fetch data

### ISSUE THREAD

**Issue by sharukhmohammed** [jeancsanchez/Simple-MVP-Retrofit-example#1]
CountryPresenter is currently responsible for fetching the data, which actually must be done by CountryModel. This violates the separation of concerns

**Comment by jeancsanchez** [comment]
@sharukhmohammed I agree with you, however, it is not the case here. If you are taking into account the principles that follow DDD and/or Clean Architecture, this can be the truth. But this project is a simple example of how to start with MVP. The main idea of this repository is to show for beginners how to start using MVP and they should improve this code on their own projects.
